Scripts/Core/EditorVR.Menus.cs

```diff
@@ -30,7 +30,8 @@ internal enum MenuHideFlags
 				NearWorkspace = 1 << 2,
 			}
 
-			readonly Dictionary<Type, ISettingsMenuProvider> m_SettingsMenuProviders = new Dictionary<Type, ISettingsMenuProvider>();
+			readonly Dictionary<KeyValuePair<Type, Transform>, ISettingsMenuProvider> m_SettingsMenuProviders = new Dictionary<KeyValuePair<Type, Transform>, ISettingsMenuProvider>();
```

Comment:
We're going to punt on this for now - we haven't found a solution that was convincingly more elegant than this